frontend/dockerfile/dockerfile_test.go

```diff
@@ -4767,6 +4768,36 @@ COPY Dockerfile Dockerfile
 	require.True(t, called)
 }
 
+// moby/buildkit#1301
+func testDockefileCheckHostname(t *testing.T, sb integration.Sandbox) {
+	f := getFrontend(t, sb)
+	dockerfile := []byte(`
+FROM busybox
+RUN cat /etc/hosts | grep testtest
```

Comment:
also check `hostname(1)` and `$HOSTNAME`

Reply:
done, PTAL~